test/integration/tcp_proxy_integration_test.cc

```diff
@@ -68,6 +69,87 @@ void TcpProxyIntegrationTest::initialize() {
+    access_log->set_name("accesslog");
+    envoy::extensions::access_loggers::file::v3::FileAccessLog access_log_config;
+    access_log_config.set_path(access_log_path);
+    access_log_config.mutable_log_format()->mutable_text_format_source()->set_inline_string(params);
+    access_log->mutable_typed_config()->PackFrom(access_log_config);
+    auto* runtime_filter = access_log->mutable_filter()->mutable_runtime_filter();
+    runtime_filter->set_runtime_key("unused-key");
+    auto* percent_sampled = runtime_filter->mutable_percent_sampled();
+    percent_sampled->set_numerator(100);
+    percent_sampled->set_denominator(envoy::type::v3::FractionalPercent::DenominatorType::
+                                         FractionalPercent_DenominatorType_HUNDRED);
+    config_blob->PackFrom(tcp_proxy_config);
+  });
+}
+
+inline std::string setupByteMeterAccessLog(ConfigHelper& config_helper,
+                                           Network::Address::IpVersion& version) {
+  auto file_name =
+      fmt::format("access_log{}{}.txt", version == Network::Address::IpVersion::v4 ? "v4" : "v6",
+                  TestUtility::uniqueFilename());
+  std::string access_log_path = TestEnvironment::temporaryPath(file_name);
+  setListenerAccessLog(config_helper, access_log_path,
+                       "DOWNSTREAM_WIRE_BYTES_SENT=%DOWNSTREAM_WIRE_BYTES_SENT% "
+                       "DOWNSTREAM_WIRE_BYTES_RECEIVED=%DOWNSTREAM_WIRE_BYTES_RECEIVED% "
+                       "UPSTREAM_WIRE_BYTES_SENT=%UPSTREAM_WIRE_BYTES_SENT% "
+                       "UPSTREAM_WIRE_BYTES_RECEIVED=%UPSTREAM_WIRE_BYTES_RECEIVED%\n");
+  return access_log_path;
+}
+
+// Using a set of T, test a consumer function for a true AssertionResult, or
+// timeout.
+template <typename T>
+inline absl::optional<uint64_t> waitForFunction(std::vector<T>& connection_list,
+                                                std::chrono::milliseconds connection_wait_timeout,
+                                                std::function<AssertionResult(T&)>& func) {
+  AssertionResult result = AssertionFailure();
+  int upstream_index = 0;
+  Event::TestTimeSystem::RealTimeBound bound(connection_wait_timeout);
+  // Loop over the upstreams until the call times out or an upstream request is
+  // received.
+  while (!result) {
+    upstream_index = upstream_index % connection_list.size();
+    result = func(connection_list[upstream_index]);
+    if (result) {
+      return upstream_index;
+    } else if (!bound.withinBound()) {
+      RELEASE_ASSERT(0, "Timed out waiting for new connection.");
+      break;
+    }
+    ++upstream_index;
+  }
+  RELEASE_ASSERT(result, result.message());
+  return {};
+}
+
+// Wait until the log file is populated, and read from it.
+// Access logs only get flushed to disk periodically,
+// so poll until the log is non-empty
+template <typename T> inline std::string waitForLog(T& api, const std::string& access_log_path) {
```

Comment:
I think you can use this instead

See: https://github.com/envoyproxy/envoy/blob/63505caf5117528480954c20f80a7a5e63d43522/test/integration/base_integration_test.h#L127

Reply:
Yep, replaced all calls to this.